app/jobs/import_facilities_job.rb

```diff
@@ -12,10 +12,6 @@ def perform(facilities)
       import_facilities << import_facility
     end
 
-    ActiveRecord::Base.transaction do
-      Facility.import!(import_facilities, validate: true)
-      # import! can't run callbacks, so we manually ensure that we create regions
-      import_facilities.each(&:make_region) if Flipper.enabled?(:regions_prep)
-    end
+    Facility.import_with_regions(import_facilities, validate: true)
```

Comment:
Think we need some tests that exercise code paths with and without regions prep turned on.

What happens if someone imports facilities with new blocks?  We probably should handle that in the loop above on line 8, assuming `facilities` is the attributes from a CSV?

Reply:
So, the tests for this job are dumb because the job is really dumb. It doesn't bother validating here because the validation for CSVs has already happened since the errors need to be presented to the user before the job is even kickstarted. So even if this job throws errors, you know something bad has happened in the pipeline.

This new [class](https://github.com/simpledotorg/simple-server/blob/master/app/validators/csv/facilities_validator.rb) takes care of this from this [PR](https://github.com/simpledotorg/simple-server/pull/1554) (both with and without `regions_prep`) and it correctly reports missing zones if there are any since that particular [validation](https://github.com/simpledotorg/simple-server/blob/master/app/models/facility.rb#L92) already exists on the model.